### PR TITLE
TG-6286: Internal PR: Add unit tests for com.jzy.game.engine.util.StringUtil

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
Hi,

I've analysed your code base and noticed that `com.jzy.game.engine.util.StringUtil` is not fully tested.

I've written some tests for the functions in this class with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/products).

Hopefully these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+public class StringUtilTest {
+
+  @Test
+  public void isNullOrEmptyFalse() {
+    Assert.assertEquals(false, StringUtil.isNullOrEmpty("      `a "));
+  }
+
+  @Test
+  public void isNullOrEmptyTrue() {
+    Assert.assertEquals(true, StringUtil.isNullOrEmpty(null));
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Merge `True` and `False` test methods (with empty line to split assertions)

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import com.jzy.game.engine.util.StringUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringUtilTest {
+
+  @Test
+  public void isNullOrEmptyFalse() {
+    Assert.assertEquals(false, StringUtil.isNullOrEmpty("      `a "));
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Use a different string

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
+package com.jzy.game.engine.util;
+
+import com.jzy.game.engine.util.StringUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class StringUtilTest {
+
+  @Test
+  public void isNullOrEmptyFalse() {
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Prefix test names with `test` (to match project existing tests)

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  public void isIntegerTrue() {
+    Assert.assertEquals(true, StringUtil.isInteger("-1"));
+    Assert.assertEquals(true, StringUtil.isInteger("1"));
+    Assert.assertEquals(true, StringUtil.isInteger("+1"));
+    Assert.assertEquals(true, StringUtil.isInteger("+"));
+    Assert.assertEquals(true, StringUtil.isInteger("-"));
+  }
+
+  @Test
+  public void isIntegerFalse() {
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Merge all tests into one method, (with linebreak for true/false)

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  @Test
+  public void isNullOrEmptyTrue() {
+    Assert.assertEquals(true, StringUtil.isNullOrEmpty(null));
+    Assert.assertEquals(true, StringUtil.isNullOrEmpty("      "));
+  }
+
+  @Test
+  public void isIntegerEmpty() {
+    Assert.assertEquals(false, StringUtil.isInteger(""));
+    Assert.assertEquals(false, StringUtil.isInteger(null));
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: `null` is not empty

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    Assert.assertEquals(true, StringUtil.isDouble("-"));
+    Assert.assertEquals(true, StringUtil.isDouble("+."));
+    Assert.assertEquals(true, StringUtil.isDouble("-."));
+  }
+
+  @Test
+  public void isDoubleFalse() {
+    Assert.assertEquals(false, StringUtil.isDouble("--"));
+    Assert.assertEquals(false, StringUtil.isDouble("++"));
+    Assert.assertEquals(false, StringUtil.isDouble("a"));
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Merge `isDouble` tests

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java (hunk 2 above)
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Replace all `assertEquals(boolean` with `assertBoolean(`

File: game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  @Test
+  public void lowerFirstChar() {
+    Assert.assertEquals("a!!!!!!!", StringUtil.lowerFirstChar("A!!!!!!!"));
```
Comment (freddytuxworth) [inline game-engine/src/test/java/com/jjy/game/engine/util/StringUtilTest.java]: Use test case that shows that the remaining characters are not capitalized (capital ! == ! so this is not shown)

**Review by freddytuxworth — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Pull request closed without merging**